src/java.base/share/classes/java/lang/ScopedValue.java

```diff
@@ -532,15 +488,33 @@ private void runWith(Snapshot newSnapshot, Runnable op) {
         }
     }
 
+    /**
+     * An operation that returns a result and may throw an exception.
+     *
+     * @param <T> result type of the operation
+     * @param <X> type of the exception thrown by the operation
+     * @since 23
+     */
+    @PreviewFeature(feature = PreviewFeature.Feature.SCOPED_VALUES)
+    @FunctionalInterface
+    public interface CallableOp<T, X extends Throwable> {
```

Comment:
I feel like that a functional interface such as this one should be promoted to a top‑level type in some common package by the time this feature exits preview.